Source/WebCore/rendering/RenderText.cpp

```diff
@@ -1329,6 +1329,55 @@ UChar RenderText::previousCharacter() const
     return previousString[previousString.length() - 1];
 }
 
+// https://www.w3.org/TR/css-text-3/#small-kana
+static String convertToFullSizeKana(const String& string)
+{
+    static const uint16_t smallKanas[] = {
+        // ぁ   ぃ      ぅ      ぇ      ぉ      っ      ゃ      ゅ      ょ
+        0x3041, 0x3043, 0x3045, 0x3047, 0x3049, 0x3063, 0x3083, 0x3085, 0x3087,
+        // ゎ   ゕ      ゖ
+        0x308E, 0x3095, 0x3096,
```

Comment:
Can we try very hard not to put these kinds of tables in WebKit? We should see if there's something in ICU that we can use. If not, we should make sure there's a feature request open on upstream ICU for it, and discuss with Apple's ICU team to see if we can come up with an interim solution.

Reply:
There is a csswg issue open about this: https://github.com/w3c/csswg-drafts/issues/8442

In the meantime, having it in WebKit seems a reasonable solution (note that Firefox is doing the same).

Reply:
ICU includes a transliterator called `Fullwidth-Halfwidth`. Is the problem with this performance, correctness, or did we just not try it?

Reply:
We are actually gonna use the ICU `Fullwidth-Halfwidth` for `text-transform: full-width`.

My understanding	of font stuff is quite limited, but I don't think this mapping from small to full-size Kana is actually defined anywhere but in the CSS Text spec 

https://www.w3.org/TR/css-text-3/#text-transform-mapping

> The mappings for [small Kana](https://www.w3.org/TR/css-text-3/#kana-small) to [full-size Kana](https://www.w3.org/TR/css-text-3/#kana-full-size) are defined in [Appendix G: Small Kana Mappings](https://www.w3.org/TR/css-text-3/#small-kana).

Reply:
I see. So surprised this is not in ICU!